src/App.tsx

```diff
@@ -1,39 +1,94 @@
 import React from 'react';
 import './App.scss';
 
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import LinearProgress from '@mui/material/LinearProgress';
+
+import '@fontsource/roboto/400.css';
+import '@fontsource/roboto/700.css';
+
+import { Clock } from './components/Clock/Clock';
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  timerId = 0;
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.addEventListener('click', this.leftClickHandler);
+    document.addEventListener('contextmenu', this.rightClickHandler);
+  }
 
-        {' time is '}
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+    document.removeEventListener('click', this.leftClickHandler);
+    document.removeEventListener('contextmenu', this.rightClickHandler);
+  }
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
-      </div>
-    </div>
-  );
-};
+  leftClickHandler = () => {
+    this.setState({ hasClock: true });
+  };
+
+  rightClickHandler = (event: MouseEvent) => {
+    event.preventDefault();
+
+    this.setState({ hasClock: false });
+  };
```

Comment:
p. 3 of checklist: 
"Method's name should start with a verb (so you could easily tell what your method actually do)"